client/Mac/MRDPView.m

```diff
@@ -769,7 +769,8 @@ - (void) onPasteboardTimerFired :(NSTimer*) timer
 			formatId = ClipboardRegisterFormat(mfc->clipboard, "UTF8_STRING");
 			size = (UINT32) [formatData length];
 			data = [formatData bytes];
-			ClipboardSetData(mfc->clipboard, formatId, data, size);
+			/* size is the string length without the terminating NULL terminator */
+			ClipboardSetData(mfc->clipboard, formatId, data, size + 1);
```

Comment:
Are you sure data is pointing to a buffer with a valid length of ```size + 1```? If the trailing zero would be part of formatData I'd expect ```[formatData length]``` to return the size including the trailing \0.

Reply:
@mfleisz `[formatData length]` behaves like `strlen` and returns the length without the trailing zero.